llm/validation.go

```diff
@@ -0,0 +1,63 @@
+// validation.go
```

Comment:
A few things to wrap this up and I think we're good!

1. Rename this file kv_cache.go

2. Let's make one function called `KVCacheQuantization` that will get the kv cache quantization based on env vars and the model data:

```
func KVCacheQuantization(ggml model, gpus discover.GpuInfoList) string {
    kvct := envconfig.KvCacheType()

    // check flash attention otherwise return default
   ...

    // kv cache quantization 
    // For embedding models, only allow f16 and f32
    if isEmbeddingModel && cacheType != "f16" && cacheType != "f32" {
        slog.Warn("only f16 and f32 cache types are supported for embedding models, defaulting to f16", "type", cacheType)
        return "f16"
    }

    if !slices.Contains(ValidKVCacheTypes, cacheType) {
        slog.Warn("invalid cache type, defaulting to f16", "type", cacheType)
        return "f16"
    }

    return kvct
}
```

3. Nice work with the tests!

Reply:
I like this, thanks. Implemented!